packages/dd-trace/src/aiguard/sdk.js

```diff
@@ -0,0 +1,164 @@
+  constructor (message, opts = {}) {
+    super(message)
+    this.name = 'AIGuardClientError'
+    if (opts.errors) {
+      this.errors = opts.errors
+    }
+    if (opts.cause) {
+      this.cause = opts.cause
+    }
+  }
+}
+
+class AIGuard extends NoopAIGuard {
+  constructor (tracer, config) {
+    super()
+
+    if (!config.apiKey || !config.appKey) {
+      log.error('AIGuard: missing api and/or app keys, use env DD_API_KEY and DD_APP_KEY')
+      this._initialized = false
+      return
+    }
+    this._tracer = tracer
+    this._headers = {
+      'DD-API-KEY': config.apiKey,
+      'DD-APPLICATION-KEY': config.appKey,
+    }
+    const endpoint = config.experimental.aiguard.endpoint || `https://app.${config.site}/api/v2/ai-guard`
+    this._evaluateUrl = `${endpoint}/evaluate`
+    this._timeout = config.experimental.aiguard.timeout
+    this._maxMessagesLength = config.experimental.aiguard.maxMessagesLength
+    this._maxContentSize = config.experimental.aiguard.maxContentSize
+    this._meta = { service: config.service, env: config.env }
+    this._initialized = true
+  }
+
+  #truncate (messages) {
+    const size = Math.min(messages.length, this._maxMessagesLength)
+    const result = messages.slice(-size)
+
+    for (let i = 0; i < size; i++) {
+      const message = result[i]
+      if (message.content?.length > this._maxContentSize) {
+        result[i] = { ...message, content: message.content.slice(0, this._maxContentSize) }
+      }
+    }
+    return result
+  }
+
+  #isToolCall (message) {
+    return message.tool_calls || message.tool_call_id
+  }
+
+  #getToolName (message, history) {
+    // 1. assistant message with tool calls
+    if (message.tool_calls) {
+      const names = message.tool_calls.map((tool) => tool.function.name)
+      return names.length === 0 ? null : names.join(',')
+    }
+    // 2. assistant message with tool output (search the linked tool call in reverse order)
+    const id = message.tool_call_id
+    for (let i = history.length - 2; i >= 0; i--) {
+      const item = history[i]
+      if (item.tool_calls) {
+        for (const toolCall of item.tool_calls) {
+          if (toolCall.id === id) {
+            return toolCall.function.name
+          }
+        }
+      }
+    }
+    return null
+  }
+
+  evaluate (messages, opts) {
+    if (!this._initialized) {
+      return super.evaluate(messages, opts)
+    }
+    const { block = false } = opts ?? {}
+    return this._tracer.trace(AI_GUARD_RESOURCE, {}, async (span) => {
+      const last = messages[messages.length - 1]
+      const target = this.#isToolCall(last) ? 'tool' : 'prompt'
+      span.setTag(AI_GUARD_TARGET_TAG_KEY, target)
+      if (target === 'tool') {
+        const name = this.#getToolName(last, messages)
+        if (name) {
+          span.setTag(AI_GUARD_TOOL_NAME_TAG_KEY, name)
+        }
+      }
+      span.meta_struct = {
+        [AI_GUARD_META_STRUCT_KEY]: {
+          messages: this.#truncate(messages)
+        }
```

Comment:
maybe a dumb q but i was grepping in the codebase and saw some other usage of `span.meta_struct`. does it make sense to add on to any other things that might already be there instead of overriding? or does it make sense to just set it as you already do?

```suggestion
      span.meta_struct ??= {}
      span.meta_struct[AI_GUARD_META_STRUCT_KEY] = {
        messages: this.#truncate(messages)
      }
```